### PR TITLE
Fixed typescript definition

### LINKED ISSUE #119: Current proxy store typescript definition is incompatible with Redux
The current typescript definition and store is not compatible with typed Redux libraries that expect the store to have a "replaceReducer" method.

Example error w/ typescript:

```
ERROR in [at-loader] ./onboarding/src/index.tsx:18:15 
    TS2322: Type '{ store: Store<{}>; children: Element; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Provider> & Readonly<{ children?: ReactNode; }> & ...'.
  Type '{ store: Store<{}>; children: Element; }' is not assignable to type 'Readonly<ProviderProps>'.
    Types of property 'store' are incompatible.
      Type 'Store<{}>' is not assignable to type 'Store<any>'.
        Property 'replaceReducer' is missing in type 'Store<{}>'.
```

### PR DESCRIPTION
**Pull request by tshaddix**
The typescript definition did not play with other typescript libraries such as `react-redux`. This was due to an incomplete store definition, missing interface requirements, and some miss-typed methods.

Fixes #119

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
   },
   "homepage": "https://github.com/tshaddix/react-chrome-redux#readme",
   "dependencies": {
-    "lodash": "^4.5.1"
+    "lodash": "^4.5.1",
+    "redux": "^3.7.2"
```
Comment (tshaddix) [inline package.json]: This was required for the typescript definition "import" statement to work. Followed example from `@types/react-redux`

File: index.d.ts [hunk 2]
```diff
@@ -35,26 +35,36 @@ export class Store<T> {
...
-   * @param  data The action data to dispatch
-   * @return Promise that will resolve/reject based on the action response from the background
+   * @param data The action data to dispatch
+   * 
+   * Note: Although the return type is specified as the action, react-chrome-redux will
+   * wrap the result in a responsePromise that will resolve/reject based on the
+   * action response from the background page
    */
-  dispatch<R>(data: redux.AnyAction): Promise<R>;
+  dispatch<A>(data: A): A;
```
Comment (vhmth) [inline index.d.ts]: Nice type rename to characters that match "action" (A) and "state" (S)

File: package.json (hunk 1 above)
Comment (vhmth) [reply package.json]: ack ack!

File: src/store/Store.js [hunk 3]
```diff
@@ -123,6 +123,15 @@ class Store {
   getState() {
     return this.state;
   }
+  
+  
+  /**
+   * Stub function to stay consistent with Redux Store API. No-op.
+   * @param {object} nextReducer The reducer for the store to use instead.
+   */
+  replaceReducer(nextReducer) {
```
Comment (vhmth) [inline src/store/Store.js]: Ah good call

Comment (vhmth) [reply src/store/Store.js]: Seems like the linter is complaining about the arg not being used as well as some whitespace stuff:

https://travis-ci.org/tshaddix/react-chrome-redux/builds/323888944?utm_source=github_status&utm_medium=notification

**Comment by vhmth** [comment]
Other than linting issues:

giphy 1

^ ("No Type" by Rae Sremmurd reference)

**Comment by tshaddix** [comment]
@vhmth Thanks for the review! Great gif hahaha. Going to fix up the lint errors.

File: src/store/Store.js (hunk 3 above)
Comment (tshaddix) [reply src/store/Store.js]: Got it. Just fixed that up.

**Comment by tshaddix** [comment]
@vhmth Any idea on what semver should be for this?

**Comment by tshaddix** [comment]
I could see potential breaks if someone is using this library with typescript and suddenly has a definition that does not map cleanly to their usage.

**Comment by vhmth** [comment]
Seems like you went for a minor bump - which I totally agree with!

**Pull request merged**